bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/async/HeapSizeManager.java

```diff
@@ -63,61 +73,74 @@ public long getMaxHeapSize() {
     return maxHeapSize;
   }
 
-  public synchronized void waitUntilAllOperationsAreDone() throws InterruptedException {
+  public synchronized void flush() throws InterruptedException {
+    isFlushing.set(true);
     boolean performedWarning = false;
-    while(!pendingOperationsWithSize.isEmpty()) {
+    while (true) {
```

Comment:
hm.. why an etra cleanupFinishedOperations() before (!pendingOperationsWithSize.isEmpty()) ? I thought if pendingOperationsWithSize.isEmpty(), we don't need to do anything.

"while (!pendingOperationsWithSize.isEmpty())"  not "while (pendingOperationsWithSize.isEmpty())"